### PR TITLE
update core options and tweak makefile

### PR DESCRIPTION
**Pull request by markwkidd**
* streamline existing core options (including perma-activating some options that can safely always be on)
* add two new experimental core options to begin addressing https://github.com/libretro/mame2003-libretro/issues/318 and https://github.com/libretro/mame2003-libretro/issues/30

### CODE REVIEW

**Comment by dankcushions** [comment]
I don't think we should add the options we know crashes it, even as "experimental". That sort of thing shouldn't be in master IMO.

**Comment by markwkidd** [comment]
I like to think of it as 'exposing a regression for attention' but I'll comment that part out for now

**Comment by retro-wertz** [comment]
im not sure why those option has to be removed....

**Comment by markwkidd** [comment]
@retro-wertz are you referring to the samples and the cheats options? This PR would set them to be always on -- there is no reason for those settings to ever be disabled. If their corresponding files are in place, they will work automatically without the user needing to change a core option at all.

If there is no cheats.dat or no sample, it is handled gracefully. Might as well keep those turned on in this core.

File: src/common.c [hunk 1]
```diff
@@ -1506,10 +1506,10 @@ static int process_rom_entries(struct rom_load_data *romdata, const struct RomMo
 					while (ROMENTRY_ISCONTINUE(romp));
 
 					/* if this was the first use of this file, verify the length and CRC */
-					if (baserom)
+					if (baserom && !skip_rom_verify)
 					{
 						debugload("Verifying length (%X) and checksums\n", explength);
-						verify_length_and_hash(romdata, ROM_GETNAME(baserom), explength, ROM_GETHASHDATA(baserom));
+                        verify_length_and_hash(romdata, ROM_GETNAME(baserom), explength, ROM_GETHASHDATA(baserom));
```
Comment (joolswills) [inline src/common.c]: @markwkidd regarding indentation - this for example.

Comment (markwkidd) [reply src/common.c]: Ugh it's so ugly now. Thanks.

**Pull request merged**